lib/parsers/dpx_parser.rb

```diff
@@ -35,18 +35,23 @@ def call(io)
     w = dpx_structure.fetch(:image).fetch(:pixels_per_line)
     h = dpx_structure.fetch(:image).fetch(:lines_per_element)
 
+    display_w = w
+    display_h = h
+
     pixel_aspect_w = dpx_structure.fetch(:orientation).fetch(:horizontal_pixel_aspect)
     pixel_aspect_h = dpx_structure.fetch(:orientation).fetch(:vertical_pixel_aspect)
-    pixel_aspect = pixel_aspect_w / pixel_aspect_h.to_f
 
-    image_aspect = w / h.to_f * pixel_aspect
+    # Find display height and width based on aspect only if the file structure has pixel aspects
+    if pixel_aspect_h != 0 && pixel_aspect_w != 0
+      pixel_aspect = pixel_aspect_w / pixel_aspect_h.to_f
 
-    display_w = w
-    display_h = h
-    if image_aspect > 1
-      display_h = (display_w / image_aspect).round
-    else
-      display_w = (display_h * image_aspect).round
+      image_aspect = w / h.to_f * pixel_aspect
+
+      if image_aspect > 1
+        display_h = (display_w / image_aspect).round
+      else
+        display_w = (display_h * image_aspect).round
+      end
```

Comment:
What about adding a test to cover this fix?

Reply:
Hmm.. I thought adding the spec would cover the change. I will add a test to verify the values though

Reply:
I suggest creating a test to parse this fixture file and to expect that that are no errors in the end.

Reply:
The first test here does the same https://github.com/WeTransfer/format_parser/blob/master/spec/parsers/dpx_parser_spec.rb

Reply:
hmm you're right, this test already does what I suggested!